packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -127,7 +127,10 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
     if (event is PointerMoveEvent) {
       final VelocityTracker tracker = _velocityTrackers[event.pointer];
       assert(tracker != null);
-      tracker.addPosition(event.timeStamp, event.position);
+      // PointerMoveEvent from platform's gesture up events are positionally redundant
+      // vs the last PointerMoveEvent. Do not use in velocity calculation.
+      if (event.sourceEventType != PointerChange.up)
+        tracker.addPosition(event.timeStamp, event.position);
```

Comment:
If we originally got touchesMoved @ 0,0 -> touchesMoved @ 1,1 -> touchesEnded @ 1,1
we turned it into PointerMoveEvent @ 0,0 -> PointerMoveEvent @ 1,1 -> PointerMoveEvent @ 1,1. 

If I understand what you meant, ya, we turn the last touchesEnded into a PointerMoveEvent in class and up in sourceEventType